fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/ScalarApply2AnalyticRule.java

```diff
@@ -0,0 +1,869 @@
+        }
+
+        public ScalarOperator clone(ScalarOperator scalarOperator) {
+            return scalarOperator.accept(this, null);
+        }
+
+        public List<ScalarOperator> cloneChildren(ScalarOperator scalarOperator) {
+            return scalarOperator.getChildren().stream()
+                    .map(this::clone)
+                    .collect(Collectors.toList());
+        }
+
+        @Override
+        public ScalarOperator visit(ScalarOperator scalarOperator, Void context) {
+            // Throw exception for all unexpected situations, which will be caught in the transformer
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ScalarOperator visitConstant(ConstantOperator literal, Void context) {
+            return literal;
+        }
+
+        @Override
+        public ScalarOperator visitBetweenPredicate(BetweenPredicateOperator predicate, Void context) {
+            return new BetweenPredicateOperator(predicate.isNotBetween(), cloneChildren(predicate));
+        }
+
+        @Override
+        public ScalarOperator visitBinaryPredicate(BinaryPredicateOperator predicate, Void context) {
+            return new BinaryPredicateOperator(predicate.getBinaryType(), cloneChildren(predicate));
+        }
+
+        @Override
+        public ScalarOperator visitCompoundPredicate(CompoundPredicateOperator predicate, Void context) {
+            return new CompoundPredicateOperator(predicate.getCompoundType(), cloneChildren(predicate));
+        }
+
+        @Override
+        public ScalarOperator visitExistsPredicate(ExistsPredicateOperator predicate, Void context) {
+            return new ExistsPredicateOperator(predicate.isNotExists(), cloneChildren(predicate));
+        }
+
+        @Override
+        public ScalarOperator visitInPredicate(InPredicateOperator predicate, Void context) {
+            return new InPredicateOperator(predicate.isNotIn(), cloneChildren(predicate));
+        }
+
+        @Override
+        public ScalarOperator visitIsNullPredicate(IsNullPredicateOperator predicate, Void context) {
+            return new IsNullPredicateOperator(predicate.isNotNull(), clone(predicate.getChild(0)));
+        }
+
+        @Override
+        public ScalarOperator visitLikePredicateOperator(LikePredicateOperator predicate, Void context) {
+            return new LikePredicateOperator(predicate.getLikeType(), cloneChildren(predicate));
+        }
+
+        @Override
+        public ScalarOperator visitCall(CallOperator call, Void context) {
+            return new CallOperator(call.getFnName(), call.getType(), cloneChildren(call),
+                    call.getFunction(), call.isDistinct());
+        }
+
+        @Override
+        public ScalarOperator visitCastOperator(CastOperator operator, Void context) {
+            return new CastOperator(operator.getType(), clone(operator.getChild(0)), operator.isImplicit());
+        }
+
```

Comment:
so, why not call clone direct in visit?

Reply:
`ColumnRefOperator` has special process logic, cannot use clone here.

Reply:
`ColumnRefOperator` has special prcessing logic, so cannot use clone directly.